locations.kak

```diff
@@ -0,0 +1,175 @@
+## Grep
+
+define-command -override boost-grep -docstring "grep and select matches with <a-s>
+works best if grepcmd uses a regex flavor simlar to Kakoune's
+" -params .. %{
+  try %{
+    evaluate-commands %sh{ [ -z "$1" ] && echo fail }
+    set-register / "(?S)%arg{1}"
+    grep %arg{@}
+  } catch %{
+    execute-keys -save-regs '' *
+    evaluate-commands -save-regs l %{
+      evaluate-commands -verbatim grep -- %sh{ printf %s "$kak_main_reg_slash" }
+    }
+  }
+  boost-map-quickselect
+}
+try %{ complete-command boost-grep file }
+
+define-command -override boost-map-quickselect %{ evaluate-commands -save-regs 'l' %{
+  evaluate-commands -try-client %opt{toolsclient} %sh{
+    printf %s "map buffer=$kak_opt_locations_buffer normal <tab> '"
+    printf %s '%S^<ret><a-h>/^.*?:\d+(?::\d+)?:<ret>'
+    printf %s '<a-semicolon>l<a-l>'
+    printf %s "s$(printf %s "$kak_main_reg_slash" | sed s/"'"/"''"/g"; s/</<lt>/g")<ret>"
+    printf %s "'"
+  }
+}}
+
+## Locations
+
+declare-option -hidden str locations_buffer
+declare-option -hidden str-list locations_stack
+declare-option -docstring %{Defines location patterns for location lists.
+Default locations look like "file:line[:column][:message]"
+
+Capture groups must be:
+    1: filename
+    2: line number
+    3: optional column
+    4: optional message
+} regex locations_location_format ^\h*\K([^:\n]+):(\d+)\b(?::(\d+)\b)?(?::([^\n]+))
+
+declare-option str locations_root ''
+
+declare-option regex locations_buffer_regex \*(?:callees|callers|diagnostics|goto|find|grep|implementations|lint-output|references)\*(?:-\d+)?
+
+hook -group locations global WinDisplay %opt{locations_buffer_regex} %{
+    locations-stack-push
+}
+
+hook -group locations global GlobalSetOption '^locations_buffer=(.*)$' %{
+    # TODO: Upstream as locations-write?
+    # alias "buffer=%opt{locations_buffer}" w grep-write
+    map "buffer=%opt{locations_buffer}" normal <ret> ':locations-jump<ret>'
+}
+
+# NOTE: Only relevant if other bufer-stack systems are included
+# alias global buffers-pop locations-stack-pop
+# alias global buffer-clear locations-stack-clear
+
+define-command locations-jump %{ # from grep.kak
```

Comment:
I've been using `-override` everywhere to allow reloading the entire file.
Before moving this into Kakoune proper we want to get rid of that.

Until then we could keep `-override`;
I think it can be nice because it allows to reload whole files after a config change.
Along with some `remove-hooks` and `try` for `declare-user-mode`.

Anyway, this is really not important and I'm fine either way.
I hope that we can upstream generic locations+stack fairly soon.